pkg/config/yaml.go

```diff
@@ -255,24 +256,33 @@ func (y *YamlValue) unmarshalScalar(n *yaml.Node) error {
 	}
 	y.Wrap(v)
 
-	if l, is := IsYamlExpressionLiteral(y.Unwrap()); is { // HCL literal
-		var e Expression
-		if e, err = ParseExpression(l); err != nil {
-			// TODO: point to exact location within expression, see Diagnostic.Subject
-			return nodeToPosErr(n, err)
+	if y.Unwrap().Type() == cty.String {
+		if y.v, err = parseYamlString(y.v.AsString()); err != nil {
```

Comment:
nit: My understanding is that you can access y.v directly since you know it is a string. Would it be more correct to use y.Unwrap().AsString()